sklearnex/linear_model/tests/test_ridge.py

```diff
@@ -100,3 +106,48 @@ def test_ridge_predict_before_fit(dataframe, queue):
 
         with pytest.raises(NotFittedError):
             model.predict(X_c)
+
+
+@pytest.mark.skip(reason="Forced to use original sklearn for now.")
+def test_sklearnex_multivariate_ridge_coefs():
+    from sklearnex.linear_model import Ridge
+
+    _test_multivariate_ridge_coefficients(Ridge, random_state=0)
+
+
+@pytest.mark.skip(reason="Forced to use original sklearn for now.")
+def test_sklearnex_multivariate_ridge_alpha_shape():
+    from sklearnex.linear_model import Ridge
+
+    _test_multivariate_ridge_alpha_shape(Ridge, random_state=0)
+
+
+if daal_check_version((2025, "P", 100)):
+
+    @pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
+    @pytest.mark.parametrize("overdetermined", [True, False])
+    @pytest.mark.parametrize("alpha", [0.1, 0.5, 1.0])
+    def test_ridge_overdetermined_system(dataframe, queue, overdetermined, alpha):
+        if not overdetermined and queue and queue.sycl_device.is_gpu:
+            pytest.skip(reason="Underdetermined system is not yet supported on GPU")
```

Comment:
I'm not 100% sure how ridge is implemented, but if it uses the same code as the non-regularized linear regression, you could just enable this in the fit conditions and it should work (provided the regularization is sufficiently higher than zero for the dtype), because the regularization would make the linear systems positive.

Reply:
this is a very nice catch, thanks